### PR TITLE
[pigz] Create initial build for pigz

### PR DESCRIPTION
**Pull request by omus**
I know there are some platforms that don't work. The downloads kept stalling so I'm checking them here.

### CODE REVIEW

File: P/pigz/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+# License is embedded at the end of the README
+install_license ${WORKSPACE}/srcdir/pigz-*/README
+
+cd ${WORKSPACE}/srcdir/pigz-*
+export CPPFLAGS="-I${prefix}/include"
+make -j${nproc}
+
+# Install
+mkdir -p ${prefix}/bin
+cp pigz unpigz ${prefix}/bin/
```
Comment (giordano) [inline P/pigz/build_tarballs.jl:23]:
```suggestion
export CPPFLAGS="-I${includedir}"
make -j${nproc}

# Install
mkdir -p ${bindir}
for bin in pigz unpigz; do
    cp "${bin}" "${bindir}/${exe}${exeext}"
done
```

File: P/pigz/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+install_license ${WORKSPACE}/srcdir/pigz-*/README
+
+cd ${WORKSPACE}/srcdir/pigz-*
+export CPPFLAGS="-I${includedir}"
+make -j${nproc}
+
+# Install
+mkdir -p ${bindir}
+for bin in pigz unpigz; do
+    cp "${bin}" "${bindir}/${exe}${exeext}"
```
Comment (giordano) [inline P/pigz/build_tarballs.jl:24]:
```suggestion
    cp "${bin}" "${bindir}/${bin}${exeext}"
```
:facepalm:

**Comment by omus** [comment]
FreeBSD failure:
```
[23:32:23]  ---> make -j${nproc}
[23:32:23] pigz.c: In function ‘copymeta’:
[23:32:23] pigz.c:3815:47: error: ‘S_IFMT’ undeclared (first use in this function)
[23:32:23]      if (stat(from, &st) != 0 || (st.st_mode & S_IFMT) != S_IFREG)
[23:32:23]                                                ^
[23:32:23] pigz.c:3815:47: note: each undeclared identifier is reported only once for each function it appears in
[23:32:23] pigz.c:3815:58: error: ‘S_IFREG’ undeclared (first use in this function)
[23:32:23]      if (stat(from, &st) != 0 || (st.st_mode & S_IFMT) != S_IFREG)
[23:32:23]                                                           ^
[23:32:23] pigz.c: In function ‘process’:
[23:32:23] pigz.c:3919:27: error: ‘S_IFMT’ undeclared (first use in this function)
[23:32:23]          if ((st.st_mode & S_IFMT) != S_IFREG &&
[23:32:23]                            ^
[23:32:23] pigz.c:3919:38: error: ‘S_IFREG’ undeclared (first use in this function)
[23:32:23]          if ((st.st_mode & S_IFMT) != S_IFREG &&
[23:32:23]                                       ^
[23:32:23] pigz.c:3920:38: error: ‘S_IFIFO’ undeclared (first use in this function)
[23:32:23]              (st.st_mode & S_IFMT) != S_IFIFO &&
[23:32:23]                                       ^
[23:32:23] pigz.c:3922:38: error: ‘S_IFDIR’ undeclared (first use in this function)
[23:32:23]              (st.st_mode & S_IFMT) != S_IFDIR) {
[23:32:23]                                       ^
[23:32:23] make: *** [<builtin>: pigz.o] Error 1
[23:32:23]  ---> make -j${nproc}
[23:32:23] Previous command exited with 2
[23:32:23] Child Process exited, exit code 2
```
From a quick search [FreeBSD defines these in `sys/stat.h`](https://www.freebsd.org/cgi/man.cgi?query=stat&sektion=2) as expected

**Comment by giordano** [comment]
Quoting from `/opt/${target}/${target}/sys-root/include/sys/stat.h`
```c
#if __XSI_VISIBLE
#define S_IFMT   0170000                /* type of file mask */
#define S_IFIFO  0010000                /* named pipe (fifo) */
#define S_IFCHR  0020000                /* character special */
#define S_IFDIR  0040000                /* directory */
#define S_IFBLK  0060000                /* block special */
#define S_IFREG  0100000                /* regular */
#define S_IFLNK  0120000                /* symbolic link */
#define S_IFSOCK 0140000                /* socket */
#define S_ISVTX  0001000                /* save swapped text even after use */
#endif
```
I guess `__XSI_VISIBLE` is either undefined or 0

**Pull request merged**